lib/chef/provider/package/yum.rb

```diff
@@ -1129,15 +1149,43 @@ def install_package(name, version)
                   end
                 end
               end
+              # methods don't count for packages we won't be touching
+              next if RPMVersion.parse(current_version_array[idx]) == RPMVersion.parse(v)
+              methods << method
+            end
+            # We could split this up into two commands if we wanted to, but
+            # for now, just don't support this.
+            if methods.uniq.length > 1
+              raise Chef::Exceptions::Package, "Multipackage rule #{name} has a mix of upgrade and downgrade packages. Cannot proceed."
```

Comment:
downgrade is a special case though. Chef doesn't even support it in apt/deb, and only supports it with special exceptions in yum. yum won't upgrade and downgrade in one command, so I felt being a bit more conservative here was the right choice for now. If people feel strongly I can make it run two commands.